src/nvim/api/deprecated.c

```diff
@@ -518,6 +518,7 @@ static int64_t convert_index(int64_t index)
 /// @return         Option Information
 Dictionary nvim_get_option_info(String name, Error *err)
   FUNC_API_SINCE(7)
+  FUNC_API_DEPRECATED_SINCE(11)
```

Comment:
Using `11` here since that was the API version when https://github.com/neovim/neovim/pull/23670 was merged, which was the PR that deprecated these functions.